src/backend/vulkan/src/window.rs

```diff
@@ -111,26 +182,15 @@ impl Instance {
                 .expect("XcbSurface::create_xcb_surface_khr() failed")
         };
 
-        let (width, height) = unsafe {
-            use std::mem;
-            use xcb::{Connection, xproto};
-            let conn = Connection::from_raw_conn(connection as _);
-            let geometry = xproto::get_geometry(&conn, window)
-                .get_reply()
-                .expect("xcb_get_geometry failed")
-                .ptr
-                .as_ref()
-                .expect("unexpected NULL XCB geometry");
-            mem::forget(conn); //TODO: use `into_raw_conn`
-            (geometry.width as _, geometry.height as _)
-        };
-
-        self.create_surface_from_vk_surface_khr(surface, width, height, 1)
+        self.create_surface_from_vk_surface_khr(surface, 1, Connection::Xcb(connection, window))
     }
 
     #[cfg(all(unix, not(target_os = "android")))]
     pub fn create_surface_from_wayland(
-        &self, display: *mut c_void, surface: *mut c_void, width: Size, height: Size
+        &self,
+        display: *mut c_void,
+        surface: *mut c_void,
+        window: &Arc<winit::Window>,
```

Comment:
We can't just require the user to provide a reference to a winit window as this interface is supposed to work with other crates or even C application as well.